### PR TITLE
Added TechDocs examples and related changes

### PR DESCRIPTION
**Pull request by awanlin**
This adds several examples of TechDocs usages - lists, iframe, PlantUML, and Mermaid. It also implements a hybrid build strategy so we can show both the out of the box TechDocs setup and a CI/CD with Cloud Storage example.

### CODE REVIEW

**Comment by awanlin** [comment]
Moving to Draft for now as I need to test the workflow changes. Thought I could from this branch but it's from my fork

**Comment by awanlin** [comment]
Tested and ready to go! 🚀

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+site_name: 'Backstage Demo'
+
+plugins:
+  - techdocs-core
+  - kroki
```
Comment (freben) [inline mkdocs.yml:5]: newline! :D

**Review by freben — APPROVED** [review]
Alright! One more rebase

File: mkdocs.yml (hunk 1 above)
Comment (awanlin) [reply mkdocs.yml]: Fixed!

**Pull request merged**